### PR TITLE
chore(semver): add regexp capturing groups

### PR DESCRIPTION
**Pull request by timreichen**
adds regexp capturing groups for more clarity.

### CODE REVIEW

File: semver/parse.ts [hunk 1]
```diff
@@ -37,15 +37,14 @@ export function parse(version: string | SemVer): SemVer {
 
   version = version.trim();
 
-  const m = version.match(FULL_REGEXP);
-  if (!m) {
-    throw new TypeError(`Invalid Version: ${version}`);
-  }
+  const match = version.match(FULL_REGEXP);
+  const groups = match?.groups;
+  if (!groups) throw new TypeError(`Invalid Version: ${version}`);
```
Comment (iuioiua) [inline semver/parse.ts:42]:
```suggestion
  const groups = version.match(FULL_REGEXP)?.groups;
  if (!groups) throw new TypeError(`Invalid Version: ${version}`);
```

File: semver/parse_range.ts [hunk 2]
```diff
@@ -285,8 +288,8 @@ export function parseRange(range: string): SemVerRange {
     .trim()
     .split(/\s*\|\|\s*/)
     .map((range) => {
-      // convert `1.2.3 - 1.2.4` into `>=1.2.3 <=1.2.4`
-      range = range.replace(HYPHENRANGE_REGEXP, hyphenReplace);
+      range = hyphenReplace(range);
+      // range = range.replace(HYPHENRANGE_REGEXP, hyphenReplace);
```
Comment (iuioiua) [inline semver/parse_range.ts:292]:
```suggestion
```

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Review by iuioiua — CHANGES_REQUESTED** [review]
Nice improvement in readability. Few nits.

**Review by iuioiua — APPROVED** [review]
LGTM. Nice!

**Review by iuioiua — APPROVED** [review]
LGTM. Nice!

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**